### ISSUE TITLE
x/website,x/pkgsite: add support for syntax highligting

### ISSUE THREAD

**Issue by hyangah** [golang/go#56081]
This is an umbrella issue for syntax highlighting support in various web properties owned and maintained by the Go project

- [ ] x/pkgsite: code block in comments https://github.com/golang/go/issues/40952
       example: https://pkg.go.dev/cloud.google.com/go/storage?tab=doc#hdr-Objects
- [ ] x/website: docs and tutorials https://github.com/golang/go/issues/55886
        examples:
        - https://go.dev/doc/tutorial/database-access
        - https://go.dev/doc/effective_go
- [ ] x/playground, x/pkgsite: https://github.com/golang/go/issues/34227 https://github.com/golang/go/issues/52198 https://github.com/golang/go/issues/28539 
        examples:
        - example code https://pkg.go.dev/io#example-Copy
- [ ] x/tour: tour partially supports syntax highlighting but that's unmaintained. Recent language updates aren't reflected.
       See https://github.com/golang/go/issues/34227#issuecomment-530248120 for context.

**Comment by danawoodman** [comment]
I did a brief experiment to add syntax highlighting via https://github.com/yuin/goldmark-highlighting and it seems fairly easy to accomplish. Is this something the team is open to?

<img width="675" alt="Screenshot 2023-09-02 at 4 26 01 PM" src="https://github.com/golang/go/assets/157695/ba951623-8ceb-48e1-a751-5ece777fbfcf">

**Comment by rsc** [comment]
I agree. Please do not add syntax highlighting to pkgsite and the website. This was decided a long time ago and probably not worth revisiting. Thank you!

**Comment by ufukty** [comment]
Well I don't put a vote here in either side, just here to elaborate arguments.

I don't understand why authors [choose](https://groups.google.com/g/golang-nuts/c/hJHCAaiL0so/m/E2mQ1RDiio8J) to ignore community requests that completely aligns with industry wide practices by arguing the feature is only needed/prioritized by unexperienced users(?). Otherwise members of r/golang choosing alternative playgrounds* that support syntax highlighting to share examples perfectly indicates the direction of community vote. I expect the lack of beginner friendly UI to create unwelcoming onboarding experience for the new developers, which might be a valid concern.

For accessibility related concerns, many of the modern operating systems provide "color filter" options each are optimized for different variation of color vision deficiencies. eg. [iOS](https://support.apple.com/en-bw/guide/iphone/iph3e2e1fb0/18.0/ios/18.0), [Android](https://support.google.com/accessibility/android/answer/11183305?hl=en&ref_topic=9079043&sjid=15769352192097505038-EU), [macOS](https://support.apple.com/en-bw/guide/mac-help/mchl11ddd4b3/mac), [Windows](https://support.microsoft.com/en-us/windows/use-color-filters-in-windows-43893e44-b8b3-2e27-1a29-b0c15ef0e5ce). On top of that, adding a persistent toggle is completely fine and trivial.

For the implementation difficulty and frontend performance, the package site has the advantage of content being static. Thus, the to-HTML conversion can be performed server-side. There are plenty of packages for that such as [alecthomas/chroma](https://github.com/alecthomas/chroma) which is working fine in my documentation site project. Although it doesn't support LSP provided information. The variety of tokens is pretty limited, and the quality is incomparable with LSP.

\*alternative playgrounds: 
- [goplay.tools](https://goplay.tools) (monaco based)
- [goplay.space](https://goplay.space)

**Comment by danawoodman** [comment]
@ufukty explained it clearly and eloquently

is the team open to a community poll on this?

i would assume the vast majority of end users of go would appreciate syntax highlighting to help them read and understand go code in the docs, REPL etc, but a poll could prove it either way

i know that go is famous for saying "no" to features and bloat (which we all deeply appreciate), but something that makes go more approachable to most users should be something worth seriously considering, especially if backed up by a community vote

**Comment by VoxelPrismatic** [comment]
Well, I just added TreeSitter syntax highlighting to `go doc` https://github.com/VoxelPrismatic/godoc/

![Image](https://github.com/user-attachments/assets/58e5aa72-29de-403b-8e59-e5d4587665d3)

![Image](https://github.com/user-attachments/assets/a62f44d4-fc7d-4656-b949-20fe1a1995e0)

**Comment by pjebs** [comment]
I prefer the top one:

# Github:

<img width="844" height="622" alt="Image" src="https://github.com/user-attachments/assets/7d01f1a7-0668-464d-b700-7c321ff570b5" />

# pkgsite:

<img width="881" height="607" alt="Image" src="https://github.com/user-attachments/assets/5e0164f1-59ab-4a8e-8931-5353478777b8" />

**Comment by ufukty** [comment]
Even differentiating the style of comment lines would contribute to overall readability significantly before going further.
